src/Parlot/Character.cs

```diff
@@ -26,19 +26,24 @@ public static bool IsIdentifierPart(char ch)
         public static bool IsWhiteSpace(char ch)
         {
             return (ch == 32) || // space
-                   (ch == '\t') || // horizontal tab
-                   (ch == 0xC) || // form feed - new page
+                   (ch == 9) || // horizontal tab
                    (ch == 0xA0) || // non-breaking space
-                   (ch >= 0x1680 && (
-                                        ch == 0x1680 ||
-                                        ch == 0x180E ||
-                                        (ch >= 0x2000 && ch <= 0x200A) ||
-                                        ch == 0x202F ||
-                                        ch == 0x205F ||
-                                        ch == 0x3000 ||
-                                        ch == 0xFEFF));
+                   IsWhiteSpaceNonAscii(ch);
```

Comment:
If we consider `IsWhiteSpaceNonAscii` a very rare case to happen, I would probably change this to: `|| (ch >= 0x1680 && IsWhiteSpaceNonAscii(ch)` and mark `IsWhiteSpaceNonAscii` with no inlining. This would stop the check in `IsWhiteSpace` and prevent the method call, making `IsWhiteSpace` smaller (and better inlining candidate).

If this `IsWhiteSpaceNonAscii` is common call, it could be with aggressive inlining with method body `return ch >= 0x1680 && IsWhiteSpaceNonAsciUnlikely()` which is a call to method with no inlining.